docs/spec_draft.md

```diff
@@ -1263,6 +1265,116 @@ Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.sort
+
+### Semantics
+
+Sorts a variadic number of tensors in `inputs` together, according to a custom 
+`comparator`, along the given `dimension` and produces a variadic number of
+tensors as `result`. If `is_stable` is set to true, the sort is guaranteed to be
+stable, that is, if there are elements which are considered to be equal by the
+comparator, the relative order of the equal values is preserved.
+
+If only one operand is provided:
+
+* If the operand is a rank-1 tensor (an array), the result is a sorted array.
+  If you want to sort the array into ascending order, the comparator should
+  perform a less-than comparison. Formally, after the array is sorted, it holds
+  for all index positions `i, j` with `i < j` that either
+  `comparator(value[i], value[j]) = comparator(value[j], value[i]) = false` or
+  `comparator(value[i], value[j]) = true`.
+
+* If the operand has higher rank, the operand is sorted along the provided
+  dimension. For example, for a rank-2 tensor (a matrix), a dimension value of
+  `0` will independently sort every column, and a dimension value of `1` will
+  independently sort each row. If no dimension number is provided, then the last
+  dimension is chosen by default. For the dimension which is sorted, the same
+  sorting order applies as in the rank-1 case.
+
+If `n > 1` operands are provided:
+
+* All `n` operands must be tensors with the same dimensions. The element types
+  of the tensors may be different.
+
+* All operands are sorted together, not individually. Conceptually the operands
+  are treated as a tuple. When checking whether the elements of each operand at
+  index positions `i` and `j` need to be swapped, the comparator is called with
+  `2 * n` scalar parameters, where parameter `2 * k` corresponds to the value at
+  position `i` from the `k-th` operand, and parameter `2 * k + 1` corresponds to
+  the value at position `j` from the `k-th` operand. Usually, the comparator
+  would thus compare parameters `2 * k` and `2 * k + 1` with each other and
+  possibly use other parameter pairs as tie breakers.
+
+* The result is a tuple that consists of the operands in sorted order (along
+  the provided dimension, as above). The `i-th` operand of the tuple corresponds
+  to the `i-th` operand of Sort.
```

Comment:
I think that all these bullets can be compressed as follows:

```
More formally, for all `0 <= id < jd < R`, either 
`compare_i_j = compare_j_i = false` or `compare_i_j = true`, where:
  1. `compare_i_j` $=$ `comparator(inputs[0][i], inputs[0][j], inputs[1][i], inputs[1][j], ...)`.
  1. For all indices `i = [i0, ..., iR-1]` and `j = [j0, ..., jR-1]`.
  1. Where `i` $=$ `j` everywhere except for the `d`th dimension.
  1. Where `d` $=$ `dimension >= 0 ? dimension : rank(inputs[0]) + dimension`.
```

Reply:
Thanks a lot, changed this to `0 <= id < jd < R`   ->  `0 <= id < jd < di`. Everything else being same.
Introducing subscript in #289 should make things better in this regard.